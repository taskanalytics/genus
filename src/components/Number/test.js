import React from 'react'
import { render } from 'react-testing-library'

import NumberComponent from './index.js'

describe('<Number> component', () => {
  it('renders with just value as span', () => {
    const { container } = render(<NumberComponent value={10} />)
    const node = container.firstChild
    expect(node).toMatchSnapshot()
    expect(node.tagName).toBe('SPAN')
    expect(node.textContent).toBe('10')
  })

  it('renders with description as dl', () => {
    const { container } = render(
      <NumberComponent description='Test' value={10} />,
    )
    const node = container.firstChild
    expect(node).toMatchSnapshot()
    expect(node.tagName).toBe('DL')
    expect(node.firstChild.tagName).toBe('DT')
    expect(node.firstChild.textContent).toBe('10')
    expect(node.lastChild.tagName).toBe('DD')
    expect(node.lastChild.textContent).toBe('Test')
  })
})
