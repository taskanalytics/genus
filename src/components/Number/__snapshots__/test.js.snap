// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Number> component renders with description as dl 1`] = `
<dl
  class="css-daetez"
  data-genus="Number"
>
  <dt
    class="css-9ob1md"
  >
    10
  </dt>
  <dd
    class="css-50zxo5"
  >
    Test
  </dd>
</dl>
`;

exports[`<Number> component renders with just value as span 1`] = `
<span
  class="css-9ob1md"
  data-genus="Number"
>
  10
</span>
`;
